### PR TITLE
Fix memory leak

### PR DESCRIPTION
**Pull request by dmitriie-brd**

### CODE REVIEW

File: lib/common/options.js [hunk 1]
```diff
@@ -296,7 +296,7 @@ Options.defaults = {
   },
   watchers: {
     webpack: {
-      ignored: /-dll/
+      ignored: /(-dll)|(nuxt\.config\.js)|(\.nuxt\/)/
```
Comment (0pt1m1z3r) [inline lib/common/options.js]: I think instead ".nuxt" need to use option buildDir

Comment (0pt1m1z3r) [reply lib/common/options.js]: What is the best way to do this?

**Review by atinux — COMMENTED** [review]
I am not sure about ignoring .nuxt/ since .nuxt/router.js is updated everytime we add or remove a file in pages/

@pi0, do you agree?

**Comment by 0pt1m1z3r** [comment]
New version:
```js
      ignored: [
        /-dll/,
        /\/nuxt\.config\.js$/,
        function (path) {
          // ignore all in .nuxt, exclude router.js
          return path.indexOf('/.nuxt/') !== -1 && path.indexOf('/.nuxt/router.js') === -1
        }
      ]

```

**Comment by 0pt1m1z3r** [comment]
But it return "double compiling"

**Comment by pi0** [comment]
I agree with @atinux simply ignoring .nuxt makes dev mode unable to reload on route changes. Maybe it can fixed if we gave two watchers one for distDir (.nuxt) to rebuild only if previous was build was not for regenerate:

- A new file added to srcDir
- Watcher 1 triggers and do build(gen=true)
- Build generates new .nuxt
- Watcher 2 for . nuxt triggers (current bug)
- Don't rebuild for watcher 1 as previous one was gen=true

**Comment by pi0** [comment]
I suggest moving ignore to another PR so we can discuss more.

**Comment by 0pt1m1z3r** [comment]
Returned the old value

**Review by pi0 — APPROVED** [review]

**Comment by Atinux** [comment]
Thank you so much :)

**Pull request merged**